components/raftstore/src/store/config.rs

```diff
@@ -532,13 +533,16 @@ impl Config {
                 limit
             ));
         }
-        if let Some(size) = self.apply_batch_system.max_batch_size {
-            if size == 0 {
-                return Err(box_err!("apply-max-batch-size should be greater than 0"));
-            }
-        } else {
-            self.apply_batch_system.max_batch_size = Some(256);
+
+        if self.apply_batch_system.max_batch_size == 0 {
+            self.apply_batch_system.max_batch_size = 256;
+        }
```

Comment:
Actually, when the online config is first implemented, the `TiKvConfig::validate` function is used for validation only not for adjusting, so the configs that adjusted in `TiKvConfig::validate` won't be actually used. 

Changing
https://github.com/tikv/tikv/blob/8b86c063c21f111a2fb0d4fcc4a328c3df7bd6f3/src/config.rs#L3480-L3484
to
```rust
let diff = {
        let mut incoming = self.get_current();
        incoming.update(diff);
        // Config may be adjusted in `validate`
        incoming.validate()?;
        self.get_current().diff(&incoming)
};
```
may work.

Reply:
done.